3-Solving-Problems-By-Searching/poisson_distribution.js

```diff
@@ -0,0 +1,78 @@
+function poissonDistribution(height, width, radius) {
+  let sample = poissonDiscSampler(width, height, radius);
+  let nodes = [];
+  let s = sample();
+  let grid = [];
+  while (s) {
+    nodes.push(s.point);
+    grid = s.grid;
+    s = sample();
+  }
+  return [nodes, grid];
+}
+
+function poissonDiscSampler(width, height, radius) {
```

Comment:
It comes under gpl-3.0 license. Got it from https://gist.github.com/mbostock/dbb02448b0f93e4c82c3